src/quardle/mod.rs

```diff
@@ -0,0 +1,268 @@
+        if !std::path::Path::new(format!("{}", self.clone().get_work_dir()).as_str()).exists() {
+            remove_dir_all(self.clone().get_work_dir()).unwrap();
+        }
+        Ok(())
+    }
+
+    /// Return the path to the quardle working directory
+    /// /tmp/quark/builds/<quardle_name>/
+    /// Used to create temporary files used to build quardle
+    fn get_work_dir(self) -> String {
+        format!("{}{}/", QUARK_BUILD_DIR, self.name)
+    }
+
+    /// Setup default quark configuration 
+    /// Create a `quark` directory in /opt
+    fn setup(&self) -> &Quardle {
+        // Create the quark configuration directory
+        if !std::path::Path::new(&format!("{}", QUARK_CONFIG_DIR)).exists() {
+            println!("Quark configuration directory does not exist, creating it !");
+            Command::new("mkdir")
+                .arg("/opt/quark")
+                .output()
+                .expect("failed to setup quark");
+        }
+
+        // Install kaps sources
+        if !std::path::Path::new(&format!("{}kaps", QUARK_CONFIG_DIR)).exists() {
+            println!("Kaps not found, installing it !");
+            Command::new("git")
+                .args(["clone", "https://github.com/virt-do/kaps.git"]) // Using https protocol because it seems not supporting ssh
+                .current_dir(format!("{}", QUARK_CONFIG_DIR))
+                .output()
+                .expect("failed to fetch kaps");
+        }
+
+        // Install a default kernel configuration file
+        if !std::path::Path::new(&format!("{}linux-config-x86_64", QUARK_CONFIG_DIR)).exists() {
+            println!("Kernel config file not found, installing it !");
+            Command::new("curl")
+                .arg("https://raw.githubusercontent.com/virt-do/lab/main/do-vmm/kernel/linux-config-x86_64")
+                .arg("-O") 
+                .current_dir(format!("{}", QUARK_CONFIG_DIR))
+                .output()
+                .expect("failed to fetch kernel config file");
+        }
+
+        // Install a script to build kernel
+        if !std::path::Path::new(&format!("{}kernel", QUARK_CONFIG_DIR)).exists() {
+            println!("Kernel not found, installing and building it !");
+            Command::new("git")
+                .arg("clone")
+                .args(["--depth", "1"]) 
+                .arg("https://github.com/cloud-hypervisor/linux.git") // Using https protocol because it seems not supporting ssh
+                .args(["-b", "ch-5.14"])
+                .arg("kernel") 
+                .current_dir(format!("{}", QUARK_CONFIG_DIR))
+                .output()
+                .expect("failed to download kernel builder");
+            // Use our config file
+            Command::new("cp")
+                .arg(format!("{}linux-config-x86_64", QUARK_CONFIG_DIR))
+                .arg(format!("{}kernel/.config", QUARK_CONFIG_DIR))
+                .output()
+                .expect("failed to copy kernel config");
+
```

Comment:
You need to `make oldconfig` after the copy. Make sure it runs without needing any user interaction.

Reply:
What should be expected from `make oldconfig` ? It seems to make minor change to `.config` file.

Reply:
You need to run `make oldconfig` locally and use the generated `.config`. Or call `make olddefconfig` as it will automatically pick the defaults values.